src/celeritas/optical/model/RayleighModel.cc

```diff
@@ -19,27 +20,73 @@ namespace celeritas
 {
 namespace optical
 {
+//---------------------------------------------------------------------------//
+/*!
+ * Search for material ID corresponding to the given optical material ID.
+ *
+ * Assumes material ID to optical material ID mapping is one-to-one.
+ */
+::celeritas::MaterialId
+get_core_material(::celeritas::MaterialParams const& params,
+                  OpticalMaterialId opt_mat)
+{
+    for (auto mat : range(::celeritas::MaterialId{params.num_materials()}))
+    {
+        if (::celeritas::MaterialView(params.host_ref(), mat)
+                .optical_material_id()
+            == opt_mat)
+        {
+            return mat;
+        }
+    }
+    return ::celeritas::MaterialId{};
+}
+
 //---------------------------------------------------------------------------//
 /*!
  * Construct the model from imported data.
  */
+RayleighModel::RayleighModel(ActionId id, SPConstImported imported)
+    : Model(id, "optical-rayleigh", "interact by optical Rayleigh")
+    , imported_(ImportModelClass::rayleigh, imported)
+{
+    for (auto mat : range(OpticalMaterialId(imported_.num_materials())))
+    {
+        CELER_VALIDATE(imported_.mfp(mat),
+                       << "Rayleigh model requires imported MFP for each "
+                          "optical material");
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct the model from imported data and imported material parameters.
+ *
+ * Uses \c RayleighMfpCalculator to calculate missing imported MFPs from
+ * material parameters.
+ */
 RayleighModel::RayleighModel(
     ActionId id,
     SPConstImported imported,
     SPConstMaterials materials,
-    std::vector<OpticalRayleighMaterial> rayleigh_materials)
+    SPConstCoreMaterials core_materials,
+    std::vector<ImportOpticalRayleigh> import_rayleigh_materials)
     : Model(id, "optical-rayleigh", "interact by optical Rayleigh")
     , imported_(ImportModelClass::rayleigh, imported)
     , materials_(std::move(materials))
-    , rayleigh_materials_(std::move(rayleigh_materials))
+    , core_materials_(std::move(core_materials))
+    , import_rayleigh_materials_(std::move(import_rayleigh_materials))
```

Comment:
Since we'll be getting the `ImportOpticalRayleigh` vector directly from the `ImportData` now, I'm not sure we should be moving it... maybe store a shared pointer to it?

Reply:
`ImportData` has a vector of `ImportOpticalMaterial`s which have `ImportOpticalRayleigh` as members. So a vector of `ImportOpticalRayliegh` would have to be made separately, like in `optical::MaterialParams` for `ImportOpticalProperty`.

I've started drafting an `ImportedMaterials` class, similar to `ImportedModels`, to pass around as a shared pointer since it seems like many classes want vectors of a specific `ImportOptical...`.

Reply:
Makes sense!